### PR TITLE
Add documentation - Main Concepts - Data Storage

### PR DESCRIPTION
**Pull request by Aschen**
https://deploy-preview-1860--doc-kuzzle.netlify.app/guides/main-concepts/2-data-storage

### CODE REVIEW

File: doc/2/guides/main-concepts/2-data-storage/index.md [hunk 1]
```diff
@@ -2,8 +2,660 @@
...
+
+<!-- 
+  @todo load at startup with Kaaf
+-->
+
+## Collection Settings
+
+In addition mappings, Elasticsearch exposes [settings](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/index-modules.html#index-modules-settings) that allow to finely configure the behavior of a collection.
+
+Those settings allows to configure [custom analyzers](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/analysis-custom-analyzer.html) for example.
```
Comment (scottinet) [inline doc/2/guides/main-concepts/2-data-storage/index.md:358]:
```suggestion
These settings allow to configure [custom analyzers](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/analysis-custom-analyzer.html) for example.
```

File: doc/2/guides/main-concepts/2-data-storage/index.md [hunk 2]
```diff
@@ -2,8 +2,660 @@
...
+```
+
+::: warning
+While updating the collection settings, the collection will be [closed](https://www.elastic.co/guide/en/elasticsearch/reference/7.5/indices-close.html) until the new configuration has been applied.
+:::
+
+
+## Kuzzle Metadata
+
+Whenever a **document is created, updated or deleted**, Kuzzle will **add or update the document's metadata**. Those metadata provides information about the document's lifecycle.
```
Comment (scottinet) [inline doc/2/guides/main-concepts/2-data-storage/index.md:390]:
```suggestion
Whenever a **document is created, updated or deleted**, Kuzzle will **add or update the document's metadata**. These metadata provide information about the document's lifecycle.
```

File: doc/2/guides/main-concepts/2-data-storage/index.md [hunk 3]
```diff
@@ -2,8 +2,660 @@
...
+
+Kuzzle imposes a **limit to the number of documents that can be written by the same request**.
+
+This limit ensures that Kuzzle and Elasticsearch are not overloaded by writing too many documents at once.
+
+By default, this limit is `200` documents per request. It is possible to configure this value in the `limits.documentsWriteCount` key in the [configuration](/core/2/guides/advanced/8-configuration) file.
+
+## Read Documents
+
+Kuzzle exposes methods to read documents. There is two way of retrieving documents:
```
Comment (scottinet) [inline doc/2/guides/main-concepts/2-data-storage/index.md:553]:
```suggestion
Kuzzle exposes methods to read documents. There are two way of retrieving documents:
```

File: doc/2/guides/main-concepts/2-data-storage/index.md [hunk 4]
```diff
@@ -2,8 +2,660 @@
...
+ - by document `_id`
```
Comment (scottinet) [inline doc/2/guides/main-concepts/2-data-storage/index.md:554]:
```suggestion
 - using a document unique identifier (`_id` field)
```

File: doc/2/guides/main-concepts/2-data-storage/index.md [hunk 5]
```diff
@@ -2,8 +2,660 @@
...
+ - adding [Kuzzle Metadata](/core/2/guides/main-concepts/2-data-storage#kuzzle-metadata)
+ - triggering [Database Notifications](/core/2/guides/main-concepts/6-realtime-engine#database-notifications) 
+ - application of [Data Validation](/core/2/guides/advanced/9-data-validation) rules
+ - respect of [write limit](/core/2/guides/main-concepts/2-data-storage#some-anchor)
+
+The following actions are available:
+ - [bulk:write](/core/2/api/controllers/bulk/write): write a document
+ - [bulk:mWrite](/core/2/api/controllers/bulk/m-write): write multiple documents
+ - [bulk:import](/core/2/api/controllers/bulk/import): import documents as fast as possible
+ - [bulk:deleteByQuery](/core/2/api/controllers/bulk/write): deletes large volume of documents matching a query
```
Comment (scottinet) [inline doc/2/guides/main-concepts/2-data-storage/index.md:617]:
```suggestion
 - [bulk:deleteByQuery](/core/2/api/controllers/bulk/write): delete large volume of documents matching a query
```

File: doc/2/guides/main-concepts/2-data-storage/index.md [hunk 6]
```diff
@@ -2,8 +2,660 @@
...
+
+::: warning
+Bulk actions are intended to be used by administrator and scripts.
```
Comment (scottinet) [inline doc/2/guides/main-concepts/2-data-storage/index.md:620]:
```suggestion
Bulk actions are intended to be used by administrators and scripts.  
```

File: doc/2/guides/main-concepts/2-data-storage/index.md [hunk 7]
```diff
@@ -2,8 +2,660 @@
...
+ - [bulk:deleteByQuery](/core/2/api/controllers/bulk/write): deletes large volume of documents matching a query
+
+::: warning
+Bulk actions are intended to be used by administrator and scripts.  
+It is considered harmful to let end users execute those actions.
+:::
+
+## Integrated Elasticsearch Client
+
+Kuzzle uses and exposes [Elasticsearch Javascript SDK](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html).
```
Comment (scottinet) [inline doc/2/guides/main-concepts/2-data-storage/index.md:626]:
```suggestion
Kuzzle uses and exposes the [Elasticsearch Javascript SDK](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html). 
```

File: doc/2/guides/main-concepts/2-data-storage/index.md [hunk 8]
```diff
@@ -2,8 +2,660 @@
...
+## Integrated Elasticsearch Client
+
+Kuzzle uses and exposes [Elasticsearch Javascript SDK](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html). 
+
+It is possible to **interact directly with Elasticsearch** through clients exposed in the [Backend.storage](/core/2/some-link) property.
+
+This property offers the possibility to **instantiate a new client** or to **use a lazy-instantiated client**. In both cases, the clients are configured to use the same Elasticsearch cluster as Kuzzle.
+
+::: info
+It is possible to overload the configuration used by default by instantiating a new Ealsticsearch client with the constructor [Backend.storage.Client](/core/2/some-link).
```
Comment (scottinet) [inline doc/2/guides/main-concepts/2-data-storage/index.md:633]:
```suggestion
It is possible to overload the configuration used by default by instantiating a new Elasticsearch client with the constructor [Backend.storage.Client](/core/2/some-link).
```

File: doc/2/guides/main-concepts/2-data-storage/index.md [hunk 9]
```diff
@@ -2,8 +2,660 @@
...
+// Use directly an Elasticsearch client instance
+await app.storage.client.index(esRequest)
+
+// Instantiate and use a new client
+const esClient = new app.storage.Client()
+await esClient.index(esRequest)
+```
+
+::: warning
+Kuzzle use an [internal naming system](/core/2/guides/main-concepts/2-data-storage#some-anchor) to map Elasticsearch index names with Kuzzle indexes and collections names.
```
Comment (scottinet) [inline doc/2/guides/main-concepts/2-data-storage/index.md:660]:
```suggestion
Kuzzle uses an [internal naming system](/core/2/guides/main-concepts/2-data-storage#some-anchor) to map Elasticsearch index names with Kuzzle indexes and collections names.
```

**Review by scottinet — CHANGES_REQUESTED** [review]

**Review by scottinet — APPROVED** [review]

**Review by rolljee — APPROVED** [review]
Very clear 👍

**Pull request merged**